psi4/src/psi4/occ/postprocessing.cc

```diff
@@ -138,19 +133,14 @@ void OCCWave::mp3_postprocessing() {
 
     variables_["MP3 TOTAL ENERGY"] = Emp3;
     Process::environment.globals["SCS-MP3 TOTAL ENERGY"] = Escsmp3;
-    Process::environment.globals["SOS-MP3 TOTAL ENERGY"] = Esosmp3;
-    Process::environment.globals["SCSN-MP3 TOTAL ENERGY"] = Escsnmp3;
-    Process::environment.globals["SCS-MP3-VDW TOTAL ENERGY"] = Escsmp3vdw;
-    Process::environment.globals["SOS-PI-MP3 TOTAL ENERGY"] = Esospimp3;
 
     variables_["MP2.5 CORRELATION ENERGY"] = (Emp2 - Escf) + 0.5 * (Emp3 - Emp2);
     variables_["MP2.5 TOTAL ENERGY"] = 0.5 * (Emp3 + Emp2);
     variables_["MP3 CORRELATION ENERGY"] = Emp3 - Escf;
     Process::environment.globals["SCS-MP3 CORRELATION ENERGY"] = Escsmp3 - Escf;
-    Process::environment.globals["SOS-MP3 CORRELATION ENERGY"] = Esosmp3 - Escf;
-    Process::environment.globals["SCSN-MP3 CORRELATION ENERGY"] = Escsnmp3 - Escf;
-    Process::environment.globals["SCS-MP3-VDW CORRELATION ENERGY"] = Escsmp3vdw - Escf;
-    Process::environment.globals["SOS-PI-MP3 CORRELATION ENERGY"] = Esospimp3 - Escf;
+
+    variables_["CUSTOM SCS-MP3 CORRELATION ENERGY"] = os_scale * Emp3AB + ss_scale * (Emp3AA + Emp3BB);
+    variables_["CUSTOM SCS-MP3 TOTAL ENERGY"] = Escf + variables_["CUSTOM SCS-MP3 CORRELATION ENERGY"];
```

Comment:
I see why you always need to define the `CUSTOM ` vars c-side for your neat current energy lookups. And I guess it's worth keeping the current energy setting c-side in case we can ever apply scaling to the gradient. So could you add something like the below (untested) to proc.py so that CUSTOM variables aren't cluttering up unnecessarily?

```
if not((core.option_has_changed("OCC", "SS_SCALE") or core.option_has_changed("OCC", "SS_SCALE")):
    for qcv in wfn.variables():
        if qcv.startswith("CUSTOM SCS"):    
            wfn.del_variable(qcv)
```